hpe3parclient/client.py

```diff
@@ -913,7 +951,31 @@ def copyVolume(self, src_name, dest_name, dest_cpg, optional=None):
         # Virtual volume sets are not supported with the -online option
         parameters = {'destVolume': dest_name,
                       'destCPG': dest_cpg}
+        # For online copy, there has to be tpvv/tdvv(Non merlin array)
+        # and tpvv/reduce(merlin array) has to be passed from caller side
+        # For offline copy, parameters tpvv/tdvv/reduce are invalid,
+        # has to be taken care by caller side
         if optional:
+            if self.merlin_supported:
+                # For merlin array there is no compression and tdvv parameters,
+                # removing tdvv and replacing compression with reduce field
+                if 'tdvv' in optional:
+                    optional.pop('tdvv')
+
+                if optional.get('tpvv') and optional.get('compression'):
+                    optional['reduce'] = True
+
+                if 'compression' in optional:
+                    if optional.get('compression'):
+                        optional['reduce'] = True
+                    else:
+                        optional['reduce'] = False
+
+                    optional.pop('compression')
+
+                if optional.get('tpvv') and optional.get('reduce'):
+                    optional.pop('tpvv')
+
```

Comment:
If you agree with the above attached truth table values, please create a similar one for this flow as well.

Reply:
done